smart_open/tests/test_smart_open.py

```diff
@@ -136,17 +137,27 @@ def test_s3_invalid_uri_missing_colon(self):
             "s3://access_id@access_secret@mybucket@port/mykey",
         )
 
-    def test_webhdfs_uri(self):
-        """Do webhdfs URIs parse correctly"""
-        # valid uri, no query
-        parsed_uri = smart_open_lib._parse_uri("webhdfs://host:port/path/file")
-        self.assertEqual(parsed_uri.scheme, "webhdfs")
-        self.assertEqual(parsed_uri.uri_path, "host:port/webhdfs/v1/path/file")
-
-        # valid uri, with query
-        parsed_uri = smart_open_lib._parse_uri("webhdfs://host:port/path/file?query_part_1&query_part2")
-        self.assertEqual(parsed_uri.scheme, "webhdfs")
-        self.assertEqual(parsed_uri.uri_path, "host:port/webhdfs/v1/path/file?query_part_1&query_part2")
+    def test_webhdfs_uri_to_http(self):
+        parsed_uri = smart_open_lib._parse_uri("webhdfs://host:14000/path/file")
+        actual = webhdfs.convert_to_http_uri(parsed_uri)
+        expected = "http://host:14000/webhdfs/v1/path/file"
+        assert expected == actual
```

Comment:
> Please use `self.assertEqual`, here and everywhere:
> 
> * It's consistent with the rest of the code
> * It gives more informative messages via the test runner
It is because I'm using py.test everywhere, it is able to generate great debug info from regular assert calls (BTW I strongly recommend as unit test runner here too, it should be able to run current tests). But for now I of course change it.

Reply:
I agree that pytest is far better than the existing test runner. However, I think it's best to keep changing the test runner out of scope for this PR. We can deal with that in a separate issue later.